grails-app/domain/org/pih/warehouse/requisition/RequisitionItem.groovy

```diff
@@ -175,6 +173,15 @@ class RequisitionItem implements Comparable<RequisitionItem>, Serializable {
         )
     }
 
+    /**
+     * All picklistItems should have the same pickType in the context of RequisitionItem
+     * @return {PickType} of picklistItems belonging to requisitionItem
+     */
+    PickType pickType() {
+        // return a first value on potentially empty list of picklistItems
+        return picklistItems.find { true }?.pickType
```

Comment:
What happens if pickType is null? Do we return null down to the client and then the client knows what to default to in that situation?

If we do the migration that Kacper and Alan mentioned in a previous comment, null should only represent an error case so maybe it's not as big of a problem and we can just let the client handle it however it wants to

Reply:
I think we need to make a determination about whether we can detect the allocation method (automatic, manual) by looking at the data. I gave my analysis above but need to hear from Dariusz on whether there's a way to do it. In either case, we might not need to worry since this will become a boolean isAutoAllocated in which case we'd just need to provide a sensible default value when we cannot derive the value from existing state (probably true, but not sure).